core/iwasm/common/wasm_c_api.c

```diff
@@ -2343,7 +2353,9 @@ wasm_module_new(wasm_store_t *store, const wasm_byte_vec_t *binary)
 {
     LoadArgs args = { 0 };
     args.name = "";
-    return wasm_module_new_ex(store, binary, &args);
+    args.clone_wasm_binary = true;
+    args.wasm_binary_freeable = false;
```

Comment:
Should not set `args.wasm_binary_freeable` here, but set it in wasm_module_new_ex instead, before calling wasm_runtime_load_ex (L2302):
```C
    args.wasm_binary_freeable = !args.clone_wasm_binary;
    module_ex->module_comm_rt = wasm_runtime_load_ex(...);
```

Reply:
Yeah, I was initially thinking of doing that, but I'll have to drop `const` from `LoadArgs`, I'll do it.